### PR TITLE
Make code change for AllocationToken schema change

### PR DESCRIPTION
**Pull request by rachelguan**
This is the corresponding code change to [PR#1580: Add renewal price behavior to AllocationToken](https://github.com/google/nomulus/pull/1580). 

The enum used here is called `RenewalPriceBehavior`, which will be introduced by [PR#1573](https://github.com/google/nomulus/pull/1573). The use cases for `AllocationToken` are `DEFAULT` and `SPECIFIED`. Once PR#1573 goes in, the enum will be coming from `BillingEvent.java`.

---
This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/1581)

### CODE REVIEW

**Review by CydeWeys — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @mindhog and @rachelguan)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-MzqNOdBE_QKzyauwr_x:b8al873) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
> ```
>         redemption_domain_history_id int8,
>         redemption_domain_repo_id text,
>         renewal_price_behavior int4,
> ```

This is the wrong type.  It should be `text`

**Review by mindhog — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: all files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @rachelguan)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 131 at r2](https://reviewable.io/reviews/google/nomulus/1581#-MztixC27PaXeFs-LN4F:-MztixC27PaXeFs-LN4G:bbwzgzs) ([raw file](https://github.com/google/nomulus/blob/239f8763cab36363f40df15f9081121fbe8485e9/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L131)):*
> ```Java
>      * <p>This behavior is equivalent to renewing at standard price. Renewing a domain is either at
>      * standard price or premium price, unless the renewal price is specified. Due to the
>      * similarities between 'DEFAULT' and 'STANDARD', 'NONPREMIUM' is chosen instead. This price
> ```

It's early yet, but I'm having a hard time parsing this paragraph.  I think you can probably simplify it to something like "Causes the domain to renew at standard price even if it is a premium domain."



_Code quote:_
```Java
     * <p>This behavior is equivalent to renewing at standard price. Renewing a domain is either at
     * standard price or premium price, unless the renewal price is specified. Due to the
     * similarities between 'DEFAULT' and 'STANDARD', 'NONPREMIUM' is chosen instead. This price
     * behavior is used with anchor tenants.
```
___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-MztlPTjDO9hlJVkepG5:b-y2jz3e) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

This is the wrong type.  It should be `text`
</blockquote></details>

This is the generated schema, how did we get the wrong type for it?

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @mindhog)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 131 at r2](https://reviewable.io/reviews/google/nomulus/1581#-MztixC27PaXeFs-LN4F:-MzuJC2kDrjrfKcHjI7Q:b-lolann) ([raw file](https://github.com/google/nomulus/blob/239f8763cab36363f40df15f9081121fbe8485e9/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L131)):*
> domain
good suggestion. I'm actually going to paste this to the original PR once the wording is approved since it's not merged yet and I need the enum here for the token. 
___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-MzqXV6D5AcFD_T-Ak95:b-auqia7) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

This is the generated schema, how did we get the wrong type for it?
</blockquote></details>

not sure how this was `int4` instead of `text`` via `./nom_build generateSqlSchema`. I changed that manually. I checked [PR1580](https://github.com/google/nomulus/pull/1580/files)

**Review by CydeWeys — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @mindhog and @rachelguan)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-Mzv7j3T1RzW7FuYoXB5:bda06it) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

not sure how this was `int4` instead of `text`` via `./nom_build generateSqlSchema`. I changed that manually. I checked [PR1580](https://github.com/google/nomulus/pull/1580/files)
</blockquote></details>

What happens if you re-run `./nom_build generateSqlSchema` though?  Does the type get reverted again?  This isn't something you can change manually -- this file is generated (as it says in its name).  Indeed, I believe there's a unit test that the contents of this file equal what is actually generated from the DB.  Is that test passing now?

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 3 files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @mindhog)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-MzvTd400YgcsFe0ZLEf:b-oe15co) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

What happens if you re-run `./nom_build generateSqlSchema` though?  Does the type get reverted again?  This isn't something you can change manually -- this file is generated (as it says in its name).  Indeed, I believe there's a unit test that the contents of this file equal what is actually generated from the DB.  Is that test passing now?
</blockquote></details>

yep.. I removed that line then I ran it again. I verified it and it got changed it to `int4`.

**Review by mindhog — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @CydeWeys, @mindhog, and @rachelguan)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 131 at r2](https://reviewable.io/reviews/google/nomulus/1581#-MztixC27PaXeFs-LN4F:-MzzqPw1Egiys6djPLYS:b-vczojc) ([raw file](https://github.com/google/nomulus/blob/239f8763cab36363f40df15f9081121fbe8485e9/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L131)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

> domain
good suggestion. I'm actually going to paste this to the original PR once the wording is approved since it's not merged yet and I need the enum here for the token. 
</blockquote></details>

Actually, I found that second sentence ("Due to the similarities between 'DEFAULT' and 'STANDARD', 'NONPREMIUM' is chosen instead.") confusing as well.  Maybe just remove it? Or change it to something more like "We chose to name this "NONPREMIUM" rather than simply "STANDARD" to avoid confusion between "STANDARD" and "DEFAULT"."
___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-MzzrixWD7f7-KNXBbEB:b-r6w5rc) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

yep.. I removed that line then I ran it again. I verified it and it got changed it to `int4`. 
</blockquote></details>

Per suggestion sent in chat, I think you just need to annotate the variable with `@Enumerated(EnumType.STRING)` to get it to be generated properly.

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @mindhog)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 131 at r2](https://reviewable.io/reviews/google/nomulus/1581#-MztixC27PaXeFs-LN4F:-N--mNgR1pJGqVVtmVPC:bxg31vd) ([raw file](https://github.com/google/nomulus/blob/239f8763cab36363f40df15f9081121fbe8485e9/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L131)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

Actually, I found that second sentence ("Due to the similarities between 'DEFAULT' and 'STANDARD', 'NONPREMIUM' is chosen instead.") confusing as well.  Maybe just remove it? Or change it to something more like "We chose to name this "NONPREMIUM" rather than simply "STANDARD" to avoid confusion between "STANDARD" and "DEFAULT"."
</blockquote></details>

that sounds good ! I didn't know we could actually use first person..I was purposely changing everything to passive to avoid using first person but passive is no good neither. Thanks!
___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r1](https://reviewable.io/reviews/google/nomulus/1581#-MzqNOdAEVOPR0FIf_45:-N--mV6t2J6Kg2z_J1KV:b-m4ugp6) ([raw file](https://github.com/google/nomulus/blob/3c18b855bd1b2691008a50b3359e94062413b4fd/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

Per suggestion sent in chat, I think you just need to annotate the variable with `@Enumerated(EnumType.STRING)` to get it to be generated properly.
</blockquote></details>

yep ..not sure how the enum in the other PR worked but not this one.. fixed it and much appreciated.

**Review by CydeWeys — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @mindhog and @rachelguan)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02hDRCmVR1Fuiqs39:-N-02hDRCmVR1Fuiqs3A:b-r6tksf) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
> ```
>         redemption_domain_history_id int8,
>         redemption_domain_repo_id text,
>         renewal_price_behavior text,
> ```

Isn't this supposed to be `NOT NULL` and have a default value of `"DEFAULT"`, exactly the same as for the corresponding field on `BillingRecurrence`?  Otherwise you're going to have exactly the same problem as you would've there -- the existing data will use `null`, but newly created tokens will use `"DEFAULT"`, and then you'll have to check for both values everywhere and treat them as the same thing.

**Review by CydeWeys — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @mindhog and @rachelguan)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 110 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02xvs9agbS3tExmgh:-N-02xvs9agbS3tExmgi:brzdo26) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L110)):*
> ```Java
>   }
> 
>   // TODO(rachelguan): Remove this once PR 1573 goes in;
> ```

Don't forget to remove this before submission.

**Review by mindhog — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: all files reviewed (commit messages unreviewed), 2 unresolved discussions (waiting on @CydeWeys and @rachelguan)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02hDRCmVR1Fuiqs39:-N-3NxrsELQ7Or2KjcqF:b-kdozvo) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

Isn't this supposed to be `NOT NULL` and have a default value of `"DEFAULT"`, exactly the same as for the corresponding field on `BillingRecurrence`?  Otherwise you're going to have exactly the same problem as you would've there -- the existing data will use `null`, but newly created tokens will use `"DEFAULT"`, and then you'll have to check for both values everywhere and treat them as the same thing.
</blockquote></details>

That's actually currently defined in the golden schema for both `AllocationToken` and `BillingRecurrence`. this is the generated schema, which doesn't _necessarily_ need to correspond directly to the golden schema. 
That said, I think we can get the corresponding definition in the generated schema, see [this SO question](https://stackoverflow.com/questions/197045/setting-default-values-for-columns-in-jpa).  We should also be able to use the "nullable = False" attribute on the Column annotation to get nullable.

**Review by CydeWeys — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: all files reviewed (commit messages unreviewed), 2 unresolved discussions (waiting on @mindhog and @rachelguan)

___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02hDRCmVR1Fuiqs39:-N-3YEz_8PpOmk7igfB-:b-ddt4qv) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

That's actually currently defined in the golden schema for both `AllocationToken` and `BillingRecurrence`. this is the generated schema, which doesn't _necessarily_ need to correspond directly to the golden schema. 
That said, I think we can get the corresponding definition in the generated schema, see [this SO question](https://stackoverflow.com/questions/197045/setting-default-values-for-columns-in-jpa).  We should also be able to use the "nullable = False" attribute on the Column annotation to get nullable. 
</blockquote></details>

Yup, it looks like she missed `nullable = False`.  That would fix it.

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @mindhog)

___
*[core/src/main/java/google/registry/model/domain/token/AllocationToken.java, line 110 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02xvs9agbS3tExmgh:-N-3wRcI8cs0xAKDOEv_:b-f2nj7r) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/core/src/main/java/google/registry/model/domain/token/AllocationToken.java#L110)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

Don't forget to remove this before submission.
</blockquote></details>

I wont submit this until [PR#1573](https://github.com/google/nomulus/pull/1573) is in. 
___
*[db/src/main/resources/sql/schema/db-schema.sql.generated, line 27 at r7](https://reviewable.io/reviews/google/nomulus/1581#-N-02hDRCmVR1Fuiqs39:-N-3wdqa-iBqHTbQjqXX:b-7tfmjp) ([raw file](https://github.com/google/nomulus/blob/b31e56c81e139f22f93a66ca44124f86835f14a7/db/src/main/resources/sql/schema/db-schema.sql.generated#L27)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

Yup, it looks like she missed `nullable = False`.  That would fix it.
</blockquote></details>

added

**Review by CydeWeys — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1581#-:-N-ZvXwr8sMX_10bHVuj:bnfp4nl)* status: 0 of 4 files reviewed, all discussions resolved (waiting on @mindhog)

**Pull request merged**
